grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcService.java

```diff
@@ -221,33 +220,13 @@ public void serviceAdded(ServiceConfig cfg) throws Exception {
     }
 
     @Nullable
-    private SerializationFormat findSerializationFormat(@Nullable String contentType) {
+    private SerializationFormat findSerializationFormat(@Nullable MediaType contentType) {
         if (contentType == null) {
             return null;
         }
 
         for (SerializationFormat format : supportedSerializationFormats) {
-            for (MediaType formatType : format.mediaTypes()) {
-                if (contentType.equals(formatType.toString())) {
-                    return format;
-                }
-            }
-        }
-
-        // It's very unlikely that gRPC content types will be introduced that do not statically match our known
-        // content types, so we only bother properly parsing it if we couldn't find a match. In practice, this
-        // should never happen.
-
-        final MediaType mediaType;
-        try {
-            mediaType = MediaType.parse(contentType);
-        } catch (IllegalArgumentException e) {
-            logger.debug("Failed to parse the 'content-type' header: {}", contentType, e);
-            return null;
-        }
-
-        for (SerializationFormat format : supportedSerializationFormats) {
-            if (format.isAccepted(mediaType)) {
+            if (format.isAccepted(contentType)) {
```

Comment:
Since media type is parsed once in `DefaultPathMappingContext` and not parsed again here, this LGTM.